.github/workflows/linting.yml

```diff
@@ -0,0 +1,79 @@
+---
+on:
+  pull_request:
+    branches:
+      - '**'
+  push:
+    branches:
+      - master
+    tags:
+      - 'v**'
+
+name: Linting
+jobs:
+  code-gen:
+    name: Submariner K8s API Code Generation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@master
+      - name: Run validation
+        run: make codegen
+
+  dco:
+    runs-on: ubuntu-latest
+    name: DCO Check
+    steps:
+      - name: Get PR Commits
+        id: 'get-pr-commits'
+        uses: tim-actions/get-pr-commits@master
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: DCO Check
+        uses: tim-actions/dco@master
+        with:
+          commits: ${{ steps.get-pr-commits.outputs.commits }}
+
+  gitlint:
+    name: Gitlint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 0
+      - run: make gitlint
+
+  markdown-link-check:
+    runs-on: ubuntu-latest
+    name: Markdown Link Checks
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v1
+
+      - name: Run markdown-link-check
+        uses: gaurav-nelson/github-action-markdown-link-check@v1
+        with:
+          config-file: ".markdownlinkcheck.json"
+          check-modified-files-only: "yes"
+
+  validate:
+    name: Go and MD Linting
```

Comment:
I'd like to split these up into two different make targets and jobs (golangci-lint and markdownlint).